sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/recurrence/RecurrenceRange.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.feature.management.implementation.timewindow.recurrence;
+
+import org.springframework.util.StringUtils;
+
+import java.time.LocalDate;
+
+/**
+ * The recurrence range specifying how long the recurrence pattern repeats
+ * */
+public class RecurrenceRange {
+    /**
+     * The recurrence range type. Default value is "NoEnd"
+     * */
+    private String type = RecurrenceConstants.NO_END;
```

Comment:
Is `RecurrenceConstants.NO_END` intended to be the default value? The current way this is setup with the object mapper it isn't.

Either this is dead code, or you need to update `Recurrence ` to have a new instance of `RecurrenceRange` on line 16.

Also by doing this you should be able to remove some of the null checks in `RecurrenceEvaluator` as they'll never be null. 

You might want to do this in a few other places just to remove the null checks.

Reply:
Yes, I agree with you. Since there is a default value "NoEnd" for the range class. We should remove the null check and update the RecurrenceRange to a new instance when it is null.